PowerShell/JumpCloud Commands Gallery/Mac Commands/Mac - Install CrowdStrike Falcon Agent.md

```diff
@@ -10,23 +10,71 @@ mac
 
 ```bash
 #!/bin/bash
-DownloadUrl="https://path/to/url.pkg"
+CSBaseAddress=""
+CSClientID=""
+CSClientSecret=""
 
 ############### Do Not Edit Below This Line ###############
 
-fileName="FalconSensor.pkg"
+# API Base URL and the various endpoints we need
+oauthtoken="$CSBaseAddress/oauth2/token"
+
+# Define which version we want to get.
+# 0 means latest. 1 is N-1, 2 is N-2 and so on.
+version="0"
+
+# Now define the API query we need
+sensorlist="$CSBaseAddress/sensors/combined/installers/v1?offset=$version&limit=1&filter=platform%3A%22mac%22"
+sensordl="$CSBaseAddress/sensors/entities/download-installer/v1"
+
+# Request bearer access token using the API
+token=$( /usr/bin/curl -s -X POST "$oauthtoken" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "client_id=$CSClientID&client_secret=$CSClientSecret" )
+
+# Extract the bearer token from the json output above
+BEARER_REGEX_PATTERN='\"access_token\":\ \"(.+)\",'
+
+if [[ $token =~ $BEARER_REGEX_PATTERN ]]; then
+    bearer="${BASH_REMATCH[1]}"
+    echo "Connected to CrowdStrike"
+else
+    echo "Could not find bearer token"
+    exit 1
+fi
+
+# Work out the CrowdStrike installer, grab the SHA256 hash and use that to download that installer
+sensorv=$( /usr/bin/curl -s -X GET "$sensorlist" -H "accept: application/json" -H "authorization: Bearer $bearer" )
+
+SHA256_REGEX_PATTERN='\"sha256\":\ "([^"]*)"'
+
+if [[ $sensorv =~ $SHA256_REGEX_PATTERN ]]; then
+    sensorsha="${BASH_REMATCH[1]}"
+else
+    echo "Could not find sha256 hash"
+    exit 1
+fi
+
 # Create Temp Folder
+fileName="FalconSensor.pkg"
 DATE=$(date '+%Y-%m-%d-%H-%M-%S')
-
 TempFolder="Download-$DATE"
-
 mkdir /tmp/$TempFolder
+echo "Creating /tmp/$TempFolder..."
 
 # Navigate to Temp Folder
 cd /tmp/$TempFolder
 
-# Download File into Temp Folder
-curl -o "$fileName" "$DownloadUrl"
+# Download the client.
+echo "Beginning CrowdStrike Sensor Download..."
+download=$( /usr/bin/curl -H "Accept: application/octet-stream" -H "Authorization: bearer $bearer" -o "$fileName" "$sensordl?id=$sensorsha")
+res=$?
+
+if test "$res" != "0"; then
```

Comment:
Does this line work? I'm curious what this does.

We've got brackets on our if statements in line 49 we should use the same syntax for our conditional statements.

Reply:
```suggestion
if [[ "$res" != "0" ]]; then
```

would this be correct??